google/cloud/aiplatform/models.py

```diff
@@ -115,12 +115,44 @@ def __init__(
             credentials=credentials,
             resource_name=endpoint_name,
         )
-        self._gca_resource = self._get_gca_resource(resource_name=endpoint_name)
+
+        endpoint_name = utils.full_resource_name(
+            resource_name=endpoint_name,
+            resource_noun="endpoints",
+            project=project,
+            location=location,
+        )
+
+        # Lazy load the Endpoint gca_resource until needed
+        self._gca_resource = gca_endpoint_compat.Endpoint(name=endpoint_name)
+        self._skipped_getter_call = True
 
         self._prediction_client = self._instantiate_prediction_client(
             location=self.location, credentials=credentials,
         )
 
+    def _sync_gca_resource_if_skipped(self) -> None:
+        """Sync GAPIC service representation of Endpoint class resource only if
+        get_endpoint() was never called."""
+        if self._skipped_getter_call:
+            self._gca_resource = self._get_gca_resource(
+                resource_name=self._gca_resource.name
+            )
+            self._skipped_getter_call = False
+
+    def _sync_gca_resource(self) -> None:
+        """Sync GAPIC service representation of Endpoint resource once."""
+        if self._skipped_getter_call:
+            self._sync_gca_resource_if_skipped()
+        else:
+            super()._sync_gca_resource()
+
+    def _assert_gca_resource_is_available(self) -> None:
+        """Ensures Endpoint getter was called at least once before
+        asserting on gca_resource's availability."""
+        self._sync_gca_resource_if_skipped()
```

Comment:
I wonder if it's worth the tradeoff to introduce the `_skipped_getter_call_flag`. Would it be possible to base that state off the `_gca_resource`? For example, wouldn't a property like `create_time` only be populated if we only executed the getter?

Something like:
```self._sync_gca_resource() if not self._gca_resource.create_time```

If this is possible then we can avoid the overhead of maintaining the state flag.

Also does the order of the executed statements in this method need to be reversed. Will this still throw the informative `_assert_gca_resource_available` error:

```Python
ds = aiplatform.Dataset(..., sync=False)
job = aiplatform.CustomTrainingJob(...)
model = job.run(..., sync=False)
ep = model.deploy(..., sync=False)

print(ep.network)
```

It seems like it may throw a None attribute error from `_sync_gca_resource_if_skipped`